prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -479,7 +498,9 @@ class FHIRRouter(
             }
         }
 
-        return if (exceptionFilters.isNotEmpty()) {
+        return if (singlePass && successFilters.isNotEmpty()) {
+            Pair(true, null)
+        } else if (exceptionFilters.isNotEmpty()) {
```

Comment:
I'd prefer to check for exceptions first, just so we don't keep filters that cause exceptions without realizing it. It would result in the condition filter failing if we find an exception even if it does have a passing condition, but I think that'd be for the best.